packages/typespec-client-generator-core/lib/decorators.tsp

```diff
@@ -419,3 +419,19 @@ extern dec access(
  */
 #deprecated "@flattenProperty decorator is not recommended to use."
 extern dec flattenProperty(target: ModelProperty, scope?: valueof string);
+
+namespace Azure.ClientGenerator.Core.CSharp {
```

Comment:
This should just be CSharp here otherwise you redefine the entire structure again

Reply:
I.e the current namespace here is 
Azure.ClientGenerator.Core.Azure.ClientGenerator.Core.CSharp

Reply:
i have actually no idea how it is working right now

Reply:
I think we should have the complete namespace in the TypeSpec.
The existing definition of decorators in Azure.Core:
https://github.com/Azure/typespec-azure/blob/b8c96e8ce51b517421c424348b1d9d55d4465961/packages/typespec-azure-core/lib/decorators.tsp#L3
and 
https://github.com/Azure/typespec-azure/blob/b8c96e8ce51b517421c424348b1d9d55d4465961/packages/typespec-azure-core/lib/decorators.tsp#L139

And when we set the namespace for the implementation, it's like you suggested we should not only set the sub-namespace, like 
https://github.com/Azure/typespec-azure/blob/b8c96e8ce51b517421c424348b1d9d55d4465961/packages/typespec-azure-core/src/decorators.ts#L1524
I did the same thing in this PR.

Reply:
The example you link use block namespace not file namespace. What you have is a bug see this pr https://github.com/microsoft/typespec/pull/4050

I recommend you move it to another file if you feel like putting the whole namespace.

Reply:
I see your point now.
I would prefer to keep it nested, and update it after your fix merged.

Reply:
updated